test/parallel/test-crypto-dh.js

```diff
@@ -85,11 +85,15 @@ const crypto = require('crypto');
     }, wrongBlockLength);
   }
 
-  assert.throws(() => {
-    dh3.computeSecret('');
-  }, { message: common.hasOpenSSL3 ?
-    'error:02800080:Diffie-Hellman routines::invalid secret' :
-    'Supplied key is too small' });
+  {
+    const v = crypto.constants.OPENSSL_VERSION_NUMBER;
+    const hasOpenSSL3WithNewErrorMessage = (v >= 0x300000c0 && v <= 0x30100000) || (v >= 0x30100040 && v <= 0x30200000);
```

Comment:
I'm okay with this for now, but I wonder if it would be more future proof to invert the check so that it checks versions with the old error message. This is assuming that e.g. OpenSSL > 3.2 is more likely to be using the new error message than the old one.

Reply:
There is no guarantee for this to stay the same upstream. I checked 3.2-beta1 released today, it has the affecting commit, meaning 3.2 uses the `Supplied key is too small` message: https://github.com/openssl/openssl/commit/f3a7e6c057b5054aa05710f3d528b92e3e885268